redfish-core/lib/log_services.hpp

```diff
@@ -3407,4 +3469,1001 @@ inline void requestRoutesPostCodesEntry(App& app)
+            std::find_if(redfishUriList.begin(), redfishUriList.end(),
+                         [parentRedfishUri](const auto& ele) {
+                             return parentRedfishUri == std::get<1>(ele);
+                         });
+
+        if (parentRedfishUriIt == redfishUriList.end())
+        {
+            BMCWEB_LOG_ERROR
+                << "Failed to get parent Redfish URI DBus interface for the "
+                   "identified Redfish URI: "
+                << redfishUri
+                << " of the given DBus object path: " << dbusObjPath.str;
+            return;
+        }
+
+        crow::connections::systemBus->async_method_call(
+            [asyncResp, entryJsonIdx, dbusObjPath, uriIdPos, uriIdPattern,
+             parentRedfishUri,
+             redfishUri](const boost::system::error_code ec,
+                         const std::vector<std::string>& objects) mutable {
+                if (ec || objects.empty())
+                {
+                    BMCWEB_LOG_ERROR << "Failed to get parent DBus path for "
+                                        "identified parent Redfish URI: "
+                                     << parentRedfishUri
+                                     << " of the given DBus object path: "
+                                     << dbusObjPath.str;
+                    return;
+                }
+
+                std::string uriIdObj;
+                for (const auto& object : objects)
+                {
+                    if (dbusObjPath.str.find(object) != std::string::npos)
+                    {
+                        uriIdObj = object;
+                        break;
+                    }
+                }
+
+                if (uriIdObj.empty())
+                {
+                    BMCWEB_LOG_ERROR << "Failed to get parent DBus path for "
+                                        "identified parent Redfish URI: "
+                                     << parentRedfishUri
+                                     << " of the given DBus object path: "
+                                     << dbusObjPath.str;
+                    return;
+                }
+
+                redfishUri.replace(
+                    uriIdPos, uriIdPattern.length(),
+                    getIsolatedHwItemId(
+                        sdbusplus::message::object_path(uriIdObj)));
+
+                if (redfishUri.find(uriIdPattern) == std::string::npos)
+                {
+                    nlohmann::json& entryJson =
+                        (entryJsonIdx > 0
+                             ? asyncResp->res
+                                   .jsonValue["Members"][entryJsonIdx - 1]
+                             : asyncResp->res.jsonValue);
+
+                    entryJson["Links"]["OriginOfCondition"] = {
+                        {"@odata.id", redfishUri}};
+                }
+                else
+                {
+                    fillRedfishUriWithIds(asyncResp, entryJsonIdx, dbusObjPath,
```

Comment:
Why do we need recursive, something I have not seen anywhere else in bmcweb? Why not just a for loop ?

Reply:
How many times does this recursive call loops ?

Reply:
> How many times does this recursive call loops ?

I would day in the worst case, 
- `4 (Max recursive call) * N (No of guard records)`
  - if the user firing the `/redfish/v1/Systems/system/LogServices/HardwareIsolation/Entries`
- `4 * 1`
  - if the user firing the `/redfish/v1/Systems/system/LogServices/HardwareIsolation/Entries/n`

Reply:
Done - Using the for loop and removed the recursion function.